packages/esm-commons-lib/src/utils/encounter-list-utils.ts

```diff
@@ -106,6 +107,25 @@ export function getObsFromEncounter(
     return fetchMotherName(encounter.patient.uuid);
   }
 
+  if (type === 'visitType') {
+    return encounter.encounterType.name;
+  }
+
+  if (type === 'ageAtHivTest') {
+    return encounter.patient.age;
+  }
+
+  if (actionOptions?.formName === 'selectMchForm') {
```

Comment:
I think this should be put in the builder file where we are constructing form actions because in this `getObsFromEncounter` function, only other columns other than actions use that function, so actions can be handled in their own function. Check something I did about conditional actions in the covid lab results table, you might need something like that.

Reply:
@CynthiaKamau @hadijahkyampeire I realised the error is there in in cacx and Tb modues as well on dev